### ISSUE TITLE
Changing the model forms used in CreateWithInlinesView

### ISSUE THREAD

**Issue by timmyomahony** [AndrewIngram/django-extra-views#121]
I want to create a batch of invites. So there is a main form where I can name the batch and then an inline formset to allow the user to add 1+ email addresses. I want to control the forms used in each instance.

While it seems to make use of my custom main form, it doesn't seem to be working for the inlines. It is just showing all fields for every inline.

Here is an approximation of my setup:

**models.py**

``` python
class Batch(models.Model):
    name = models.CharField(...)

    other_field = ...

class Invite(models.Model):
    email = models.EmailField(...)
    batch = models.ForeignKey(Batch)

    other_fields = ...
```

**forms.py**

``` python
from extra_views import InlineFormSet

class BatchModelForm(forms.ModelForm):
    class Meta:
        model = Batch
        fields = ["name", ]

class InviteModelForm(forms.ModelForm):
    class Meta:
        model = Invite
        fields = ["email", ]

class InviteInlineFormSet(InlineFormSet):
    model = Invite
    form_class = InviteModelForm  # This doesn't seem to work
```

**views.py**

``` python
from .forms import InviteInlineFormSet, BatchModelForm
from .models import Batch

class BatchInviteView(CreateWithInlinesView):
    model = Batch
    form_class = BatchModelForm
    inlines = [InviteInlineFormSet, ]

    def get_success_url(self):
        ...
```

I would expect the `form_class` attribute of the `InviteInlineFormSet` to give me control over the inline model form but this doesn't seem to be the case (while `form_class = BatchModelForm` on the `BatchInviteView` is correctly giving me control of the _main_ model form class)

**Comment by jonashaag** [comment]
It's `formset_class`. Unfortunately, this is somehow missing from the documentation. If you'd volunteer to add that that'd be great.

**Comment by drvid** [comment]
I've noticed this issue as well... When trying to set form_class on InlineFormSets it's ignored, yes.
When I rename this variable to formset_class (on InlineFormSet, as is suggested above?) I get "...unexpected keyword argument 'save_as_new'"

**Comment by jonashaag** [comment]
> When I rename this variable to formset_class (on InlineFormSet, as is suggested above?) I get "...unexpected keyword argument 'save_as_new'"

Could you please post a complete traceback and a minimal reproducing example (or your entire related code)?

**Comment by apallbox** [comment]
He means straightforward `formset_class = MyModelForm` I think. 

Looks like formset ignores `form_class = MyModelForm` but it's not. Only form's `fields` being ignored so this you can fix it easy:
> 
>     def get_factory_kwargs(self):
>         kwargs = super(MyInlineFormSet, self).get_factory_kwargs()
>         kwargs.pop('fields', None)
>         return kwargs
> 

This line causes formset to ignore form's `fields` if formset's `fields` wasn't set explicitly: https://github.com/AndrewIngram/django-extra-views/blob/0.7.1/extra_views/formsets.py#L260

**Comment by jonashaag** [comment]
I still don't know how this relates to `save_as_new`, please post a traceback so it's easier to understand. Thank you.

**Comment by ad-m** [comment]
Example trackback:
```
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/utils/decorators.py", line 184, in inner
    return func(*args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/braces/views/_access.py", line 98, in dispatch
    request, *args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/contrib/auth/mixins.py", line 92, in dispatch
    return super(PermissionRequiredMixin, self).dispatch(request, *args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/extra_views/advanced.py", line 129, in get
    return super(BaseCreateWithInlinesView, self).get(request, *args, **kwargs)
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/extra_views/advanced.py", line 88, in get
    inlines = self.construct_inlines()
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/extra_views/advanced.py", line 69, in construct_inlines
    inline_formset = inline_instance.construct_formset()
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/extra_views/advanced.py", line 28, in construct_formset
    formset = super(InlineFormSet, self).construct_formset()
  File "/home/adas/.virtualenvs/bliski_publikator/local/lib/python2.7/site-packages/extra_views/formsets.py", line 40, in construct_formset
    return formset_class(**self.get_formset_kwargs())
  File "/home/adas/Devel/bliski_publikator/bliski_publikator/monitoring_pages/forms.py", line 32, in __init__
    super(MiniPageForm, self).__init__(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'save_as_new'
```

Example code:
````python
class MonitoringForm(UserKwargModelFormMixin, FormHorizontalMixin, SingleButtonMixin,
                     forms.ModelForm):
    def __init__(self, *args, **kwargs):
        super(MonitoringForm, self).__init__(*args, **kwargs)
        if not self.instance.user_id:
            self.instance.user = self.user

    class Meta:
        model = Monitoring
        fields = ['name', 'description', 'institutions', 'active']
        widgets = {
            'institutions': autocomplete.ModelSelect2Multiple(url='institutions:autocomplete'),
            'description': TinyMCE(attrs={'cols': 80, 'rows': 30})
        }


class MiniPageForm(UserKwargModelFormMixin, forms.ModelForm):
    def __init__(self, *args, **kwargs):
        super(MiniPageForm, self).__init__(*args, **kwargs)

    class Meta:
        model = Page
        fields = ['title', 'ordering']


class PageInline(InlineFormSet):
    model = Page
    fields = ['title', 'ordering']  # TODO: Use MiniPageForm in PageInline
    formset_class = MiniPageForm
    # BaseTableFormSet


class MonitoringCreateView(LoginRequiredMixin, PermissionRequiredMixin, UserFormKwargsMixin,
                           NamedFormsetsMixin, CreateWithInlinesView):
    inlines = [PageInline, ]
    inlines_names = ['pages']
    model = Monitoring
    form_class = MonitoringForm
    permission_required = 'monitorings.add_monitoring'

    def get_form_valid_message(self):
        return _("{0} created!").format(self.object)
```

**Comment by ad-m** [comment]
The following code works as expected:
```
class PageInline(InlineFormSet):
    model = Page
    fields = ['title', 'ordering']
    form_class = MiniPageForm
    formset_class = BaseTableFormSet

    def get_factory_kwargs(self):
        kwargs = super(PageInline, self).get_factory_kwargs()
        kwargs.pop('fields', None)
        return kwargs


class MonitoringCreateView(LoginRequiredMixin, PermissionRequiredMixin, UserFormKwargsMixin,
                           NamedFormsetsMixin, CreateWithInlinesView):
    inlines = [PageInline, ]
    inlines_names = ['pages']
    model = Monitoring
    form_class = MonitoringForm
    permission_required = 'monitorings.add_monitoring'

    def get_form_valid_message(self):
        return _("{0} created!").format(self.object)
```

**Comment by jonashaag** [comment]
Hm, the code @timmyomahony posted above should work: If you want to control the "main" form class, use `form_class` on the `CreateWithInlinesView` subclass; if you want to control the "inline" form class, use `form_class` on the `InlineFormSet` subclass.

**Comment by sdolemelipone** [comment]
Addressed in #178.
